resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -327,8 +333,8 @@ private[spark] class ApplicationMaster(
           ApplicationMaster.EXIT_UNCAUGHT_EXCEPTION,
           "Uncaught exception: " + StringUtils.stringifyException(e))
         if (!unregistered) {
-          unregister(finalStatus, finalMsg)
           cleanupStagingDir(stagingDir)
+          unregister(finalStatus, finalMsg)
```

Comment:
add comment saying ok to clean staging dir first because unmanaged AM can't be retried.

Reply:
DOne